### PR TITLE
Small fixes to dockerbuild

### PR DESCRIPTION
**Pull request by mfojtik**
@smarterclayton PTAL

### CODE REVIEW

**Comment by mfojtik** [comment]
@smarterclayton the pulling images one is maybe controversial, dunno if you disabled pulling on purpose. Also the progress is scrambled in the console (same problem we have in the web console). I think we should have issue to fix progress bars from Docker pull/push everywhere (special formatter that translate the RawJSONStream messages and provide eye-candy progress reporting or borrow the docker one).

File: pkg/cmd/cli/cmd/dockerbuild/dockerbuild.go [hunk 1]
```diff
@@ -67,6 +68,7 @@ func NewCmdDockerbuild(fullName string, f *clientcmd.Factory, out, errOut io.Wri
 	}
 
 	cmd.Flags().StringVar(&options.DockerfilePath, "dockerfile", options.DockerfilePath, "An optional path to a Dockerfile to use.")
+	cmd.Flags().BoolVar(&options.AllowPull, "allow-pull", true, "Pull the images that are not present.")
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/dockerbuild/dockerbuild.go]: This was a bug and has been fixed.  It should only fail to pull if you don't have auth.

File: pkg/util/docker/dockerfile/builder/dispatchers.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func maintainer(b *Builder, args []string, attributes map[string]bool, original
 	if len(args) != 1 {
 		return errExactlyOneArgument("MAINTAINER")
 	}
-
+	b.Author = args[0]
```
Comment (smarterclayton) [inline pkg/util/docker/dockerfile/builder/dispatchers.go]: If no one uses this, what's the point?

File: pkg/util/docker/dockerfile/builder/client.go [hunk 3]
```diff
@@ -191,6 +192,7 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 	}
 
 	image, err := e.Client.CommitContainer(docker.CommitContainerOptions{
+		Author:     b.Author,
```
Comment (smarterclayton) [inline pkg/util/docker/dockerfile/builder/client.go]: Move this to the other commit.

File: pkg/cmd/cli/cmd/dockerbuild/dockerbuild.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/dockerbuild/dockerbuild.go]: I'm ok with this after review - I realized we weren't setting this by default.

File: pkg/util/docker/dockerfile/builder/dispatchers.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/util/docker/dockerfile/builder/dispatchers.go]: Saw this was in the other commit.

**Comment by smarterclayton** [comment]
LGTM [merge] thanks

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**